ddtrace/utils/formats.py

```diff
@@ -89,3 +93,31 @@ def flatten_dict(d, sep=".", prefix=""):
         if isinstance(d, dict)
         else {prefix: d}
     )
+
+
+def parse_tags_str(tags_str):
+    """Parse a string of tags typically provided via environment variables.
+
+    The expected string is of the form::
+        "key1:value1,key2:value2"
+
+    :param tags_str: A string of the above form to parse tags from.
+    :return: A dict containing the tags that were parsed.
+    """
+    parsed_tags = {}
+    if not tags_str:
+        return parsed_tags
+
+    for tag in tags_str.split(","):
+        try:
+            key, value = tag.split(":", 1)
+
+            # Reject empty keys or values
+            if key == "" or value == "":
+                raise ValueError
```

Comment:
I'm pretty sure you can have empty values for tags in metrics. Are we sure it's not the case for spans?

Reply:
https://docs.datadoghq.com/tagging/#defining-tags

Reply:
> Note: A tag cannot end with a colon, for example tag:

Reply:
Do we want to validate that here too?
It'd be better to do so if the agent is going to drop tags or traces. Having an error raised is easier to debug.

Reply:
We should probably be validating on `span.set_tag`. I think I'll open up a separate PR for validating tags. Might make more sense to do it all in a consolidated place.

Reply:
agreed